producer/writer/message.go

```diff
@@ -107,3 +108,7 @@ func (m *message) ToProto(pb *msgpb.Message) {
 	m.meta.ToProto(&pb.Metadata)
 	pb.Value = m.RefCountedMessage.Bytes()
 }
+
+func (m *message) ResetProto(pb *msgpb.Message) {
```

Comment:
Hm this means you are not reusing the byte slice at all. Is the message pooled?

Reply:
The message here is pooled, I never created a byte slice in the message, it's always just referencing the bytes in the message user passed in, in the m3aggregator case, it's the bytes in the encoder.

Reply:
I see, thought the byte slice in the message is used as a buffer to decode incoming data and hence the question about reusing it, if not no worries.